### DISCUSSION TITLE
Is there a way to use Sql parameters in queries?

### DISCUSSION THREAD

**Discussion by Gillardo** [maxtoroq/DbExtensions#42] [Q&A] [answered]
For example, i have 2 insert statements that i need to execute and the second one needs to use the SCOPE_IDENTITY() created from the first insert.

Can you do this using SqlBuilder?

Something like this

```
var firstBuilder = new SqlBuilder()
    .Append("DECLARE @Id    BIGINT")
    .INSERT_INTO("test (name)")
    .VALUES("my name")
    .SELECT("@Id = SCOPE_IDENTITY())
    .AppendLine()
    .INSERT_INTO("dependent (previousId, something)")
    .VALUES("@Id", "my value");
```

**Comment by maxtoroq** [comment]
I don't know, did you try it?

**Comment by Gillardo** [comment]
Tried it but a parameter is created with that value rather than using that
parameter

**Comment by maxtoroq** [comment]
I don't follow, you are not using parameter values in your SqlBuilder, the resulting command should only contain text, what happens beyond that is out of the scope of this library.

**Comment by maxtoroq** [comment, marked as answer]
Now I see the problem, don't use the `VALUES` method, everything you pass will be treated as parameter values. Use `Append` or `AppendClause`.

**Comment by Gillardo** [comment]
Thanks. Working perfectly
